core/src/main/java/org/elasticsearch/cluster/routing/allocation/AllocationService.java

```diff
@@ -240,6 +240,28 @@ private void removeDelayMarkers(RoutingAllocation allocation) {
     }
 
     /**
+     * Reset failed allocation counter for unassigned shards
+     */
+    private void resetFailedAllocationCounter(RoutingAllocation allocation) {
+        final RoutingNodes.UnassignedShards.UnassignedIterator unassignedIterator = allocation.routingNodes().unassigned().iterator();
+        final MetaData metaData = allocation.metaData();
+        while (unassignedIterator.hasNext()) {
+            ShardRouting shardRouting = unassignedIterator.next();
+            UnassignedInfo unassignedInfo = shardRouting.unassignedInfo();
+            if (unassignedInfo.isDelayed()) {
```

Comment:
I think you copied too much. Can you look again? You would also need tests to show it works.